hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -231,4 +270,32 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tableState: HoodieTableState,
       override def close(): Unit = closeableFileGroupRecordIterator.close()
     }
   }
+
+  private def readBaseFile(file: PartitionedFile, parquetFileReader: SparkParquetReader, requestedSchema: StructType,
+                           remainingPartitionSchema: StructType, fixedPartitionIndexes: Set[Int], requiredSchema: StructType,
+                           partitionSchema: StructType, outputSchema: StructType, filters: Seq[Filter],
+                           storageConf: StorageConfiguration[Configuration]): Iterator[InternalRow] = {
+    if (remainingPartitionSchema.fields.length == partitionSchema.fields.length) {
```

Comment:
Are all of these different branches covered by the tests?

Reply:
I tested with TestSparkSqlWithCustomKeyGenerator by changing partitionColumnsToRead in HoodieHadoopFsRelationFactory  by doing:
```
//TODO: [HUDI-8098] filter for timestamp keygen columns when using custom keygen
    tableConfig.getPartitionFields.orElse(Array.empty).filter(p => p == "ts").toSeq
```
to fake what [HUDI-8036] + [HUDI-8098] will do. This exposed a case that I didn't test. For MOR with log files where we read some of the partition columns, I was not doing appending correctly. I have updated HoodieFileGroupReaderBasedParquetFileFormat.appendPartitionAndProject to do this correctly now. 

I feel like appendPartitionAndProject and readBaseFile have overlapping logic, but can't think of a better way to do this for now.

Reply:
Let's add a follow-up ticket to write proper tests covering the scenarios, after HUDI-8036 and HUDI-8098 are landed.